sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/PodInfoBuilder.java

```diff
@@ -492,6 +493,28 @@ private void setReadinessCheck(
         }
     }
 
+    private static void setTaskKillGracePeriod(
+            Protos.TaskInfo.Builder taskInfoBuilder,
+            TaskSpec taskSpec) throws InvalidRequirementException {
+        Integer taskKillGracePeriodSecondsOptional = taskSpec.getTaskKillGracePeriodSeconds();
```

Comment:
Again, thank you for the clarification to base such decisions upon whether the model field is Optional<> helps a bunch.